testsuite/features/proxy_retail_pxeboot.feature

```diff
@@ -0,0 +1,313 @@
+  Scenario: Enable Saltboot formula for hardware type group
+    Given I am on the groups page
+    When I follow "HWTYPE:Intel-Genuine" in the content area
+    And I follow "Formulas" in the content area
+    And I check the "saltboot" formula
+    And I click on "Save"
+    Then the "saltboot" formula should be checked
+
+@proxy
+@private_net
+@pxeboot_minion
+  Scenario: Parametrize the Saltboot formula
+    Given I am on the groups page
+    When I follow "HWTYPE:Intel-Genuine" in the content area
+    When I follow "Formulas" in the content area
+    And I follow first "Saltboot" in the content area
+    And I enter "disk1" in disk id field
+    And I enter "/dev/vda" in disk device field
+    And I select "msdos" in disk label field
+    And I enter "p1" in first partition id field
+    And I enter "256" in first partition size field
+    And I select "swap" in first filesystem format field
+    And I select "swap" in first partition flags field
+    And I press "Add Item" in partitions section
+    And I enter "p2" in second partition id field
+    And I enter "/" in second mount point field
+    And I select "ext4" in second filesystem format field
+    And I enter "POS_Image_JeOS6" in second OS image field
+    And I click on "Save Formula"
+    Then I should see a "Formula saved!" text
+
+@proxy
+@private_net
+@pxeboot_minion
+  Scenario: PXE boot the PXE boot minion
+    Given I am authorized as "admin" with password "admin"
+    When I reboot the PXE boot minion
+    And I wait at most 90 seconds until Salt master sees "pxeboot-minion" as "unaccepted"
+    And I accept "pxeboot-minion" key in the Salt master
+    And I navigate to "rhn/systems/Overview.do" page
+    And I wait until I see the name of "pxeboot-minion", refreshing the page
+    And I follow this "pxeboot-minion" link
+    And I wait until event "Apply states [util.syncstates, saltboot] scheduled by (none)" is completed
+    And I wait until event "Package List Refresh scheduled by (none)" is completed
+    Then the PXE boot minion should have been reformatted
+
+@proxy
+@private_net
+@pxeboot_minion
+  Scenario: Check connection from terminal to branch server
+    Given I am on the Systems overview page of this "pxeboot-minion"
+    When I follow "Details" in the content area
+    And I follow "Connection" in the content area
+    Then I should see "proxy" hostname
+
+@proxy
+@private_net
+@pxeboot_minion
+  Scenario: Install a package on the new Retail terminal
+    Given I am on the Systems overview page of this "pxeboot-minion"
+    When I install the GPG key of the server on the PXE boot minion
+    And I follow "Software" in the content area
+    And I follow "Install"
+    And I check "virgo-dummy-2.0-1.1" in the list
+    And I click on "Install Selected Packages"
+    And I click on "Confirm"
+    Then I should see a "1 package install has been scheduled" text
+    When I wait until event "Package Install/Upgrade scheduled by admin" is completed
```

Comment:
^^i You will go to the hell of Gherkins
Better use And in this last step

Reply:
No, it's an action.

Reply:
All steps are doing actions in fact :) Is just how we describe and present it to the reader.

- For me, Given is all preconditions in the test (things must be there before the action you check), in that case the action is "Install".
- When, is the action and can have some "And" steps
- Then is all the checks we do after perform the action tested, can have several "Ands" ofc, it doesn't matter if the steps are doing some "actions", if in the whole sentence what they are doing is validating our "Install" action.

Reply:
... and all steps are doing tests ;-) . As a matter of fact, you could put whatever for those keywords and it would still work. It's rather a matter of logic and of documentation.

No, it matters if some of them is "Actions". And they are not validating our install action.

The base "problem" is that our suite is big, and to keep it simple we have big scenarios. But if you think at it, each of them is made of some small sub-scenarios following the pattern `Given / When / Then`.